be/src/fs/fs_hdfs.cpp

```diff
@@ -191,9 +191,7 @@ Status HDFSWritableFile::close() {
     });
     Status st = ret->get_future().get();
     PLOG_IF(ERROR, !st.ok()) << "close " << _path << " failed";
-    if (st.ok()) {
-        _closed = true;
-    }
+    _closed = true;
```

Comment:
I think you have to consider the returned error. If the close failed, we could retry to close it and maybe succeed. If you set _closed to be true. It will lead file descriptor leak.

Reply:
https://stackoverflow.com/questions/33114152/what-to-do-if-a-posix-close-call-fails